client/packages/system/src/Patient/PatientView/PatientView.tsx

```diff
@@ -124,6 +124,13 @@ export const PatientView: FC = () => {
     filterBy: { patientId: { equalTo: patientId } },
   });
   const { patient } = usePatientCreateStore();
+  const { setCurrentPatient } = usePatientStore();
+  const { data: currentPatient } = usePatient.document.get(patientId);
+
+  useEffect(() => {
+    if (!currentPatient) return;
+    setCurrentPatient(currentPatient);
+  }, [currentPatient]);
```

Comment:
This was really annoying -- I tried to include the `setCurrentPatient(currentPatient);` in the already existing `useEffect` in the parent component (line 93 same file), but it wasn't working. I can't figure out why -- the patientId is available and being correctly passed into `const { data: currentPatient } = usePatient.document.get(patientId);` but for some reason it would always show as `undefined` within the `useEffect`, even though that `useEffect` has `currentPatient` as a dependency. And I can see in the Network tab that the `get patient` query is returning the correct data.

If anyone can see why, I'd love to know! 🤔

Reply:
Just tried it and this works for me (console output + VSCode debugger):

```typescript
 useEffect(() => {
    if (!documentName && currentPatient) {
      setDocumentName(currentPatient?.document?.name);
    }
    if (currentPatient) {
      console.log(`TEST: ${currentPatient.firstName}`);
      setCurrentPatient(currentPatient);
    }
  }, [currentPatient]);
```

Or do you mean something else?

Reply:
It works if you're on the Patient tab, and then it'll stay when you switch to other tabs. But if you start on the "Programs" tab and refresh the page, it doesn't get "currentPatient" for some reason, even though the patientId is present. 🤷‍♂️

Reply:
Ah ok I see, yes seems strange like react stops rendering it...

But I think where it is right now is actually the better place because now its in the root component and not in a tab.

Reply:
Oh yes, you're right. I thought they were the other way around for some reason. (PatientView > PatientDetailView). I guess that probably explains why it wasn't running in the Patient tab!

All good -- as it is then :)